ClientCore/Extensions/StringExtensions.cs

```diff
@@ -77,4 +79,35 @@ public static string L10N(this string defaultValue, string key, bool notify = tr
         => string.IsNullOrEmpty(defaultValue)
             ? defaultValue
             : Translation.Instance.LookUp(key, defaultValue, notify);
+
+    /// <summary>
+    /// Replace special characters with spaces in the filename to avoid conflicts with WIN32API.
+    /// </summary>
+    /// <param name="defaultValue">The default string value.</param>
+    /// <returns>File name without special characters or reserved combinations.</returns>
+    /// <remarks>
+    /// Reference: <a href="https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file">Naming Files, Paths, and Namespaces</a>.
+    /// </remarks>
+    public static string ToWin32FileName(this string filename)
+    {
+        foreach (char ch in "/\\:*?<>|")
+            filename = filename.Replace(ch, '_');
+
+        // If the user is somehow using "con" or any other filename that is
+        // reserved by WIN32API, it would be better to rename it.
+
+        HashSet<string> reservedFileNames = new HashSet<string>(new List<string>(){
+            "CON",
+            "PRN",
+            "AUX",
+            "NUL",
+            "COM1", "COM2", "COM3", "COM4", "COM5", "COM6", "COM7", "COM8", "COM9", "COM¹", "COM²", "COM³",
+            "LPT1", "LPT2", "LPT3", "LPT4", "LPT5", "LPT6", "LPT7", "LPT8", "LPT9", "LPT¹", "LPT²", "LPT³"
+        }, StringComparer.InvariantCultureIgnoreCase);
+
+        if (reservedFileNames.Contains(filename))
+            filename += "_map";
```

Comment:
@MahBoiDeveloper I just realized that as a general utility function defined in `StringExtensions` class, appending `_map` as the suffix is not appropriate as it would be an application-related behavior. I changed it to `filename += "_";`

Reply:
> @MahBoiDeveloper I just realized that as a general utility function defined in `StringExtensions` class, appending `_map` as the suffix is not appropriate as it would be an application-related behavior. I changed it to `filename += "_";`

Ok, thanks for help!